x/staking/abci.go

```diff
@@ -0,0 +1,42 @@
+package staking
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// BeginBlocker will persist the current header and validator set as a historical entry
+// and prune the oldest entry based on the HistoricalEntries parameter
+func BeginBlocker(ctx sdk.Context, k Keeper) {
+	entryNum := k.HistoricalEntries(ctx)
+	// if there is no need to persist historicalInfo, return
+	if entryNum == 0 {
+		return
+	}
+
+	// Create HistoricalInfo struct
+	lastVals := k.GetLastValidators(ctx)
+	historicalEntry := types.NewHistoricalInfo(
+		ctx.BlockHeader(),
```

Comment:
Just a question, not blocking -- why do we need the entire header? Is the height not enough?

Reply:
@cwgoes and @mossid  probably have the most context to answer this question.

Reply:
We need to be able to verify that the counterparty chain has stored the correct (past) header for our own chain.